crates/bevy_solari/src/realtime/restir_gi.wgsl

```diff
@@ -64,26 +61,24 @@ fn spatial_and_shade(@builtin(global_invocation_id) global_id: vec3<u32>) {
         gi_reservoirs_a[pixel_index] = empty_reservoir();
         return;
     }
-    let gpixel = textureLoad(gbuffer, global_id.xy, 0);
-    let world_position = reconstruct_world_position(global_id.xy, depth);
-    let world_normal = octahedral_decode(unpack_24bit_normal(gpixel.a));
-    let base_color = pow(unpack4x8unorm(gpixel.r).rgb, vec3(2.2));
-    let diffuse_brdf = base_color / PI;
+    let surface = gpixel_resolve(textureLoad(gbuffer, global_id.xy, 0), depth, global_id.xy, view.main_pass_viewport.zw, view.world_from_clip);
 
     let input_reservoir = gi_reservoirs_b[pixel_index];
-    let spatial = load_spatial_reservoir(global_id.xy, depth, world_position, world_normal, &rng);
-    let merge_result = merge_reservoirs(input_reservoir, world_position, world_normal, diffuse_brdf,
+    let spatial = load_spatial_reservoir(global_id.xy, depth, surface.world_position, surface.world_normal, &rng);
+    let merge_result = merge_reservoirs(input_reservoir, surface.world_position, surface.world_normal, surface.material.base_color / PI,
```

Comment:
If we're shading with the correct diffuse BRDF, why not use that as the canonical BRDF too?

Reply:
E.g. if we have a metallic=1 material, it will have no diffuse lobe, so neighboring diffuse pixels won't have anything to resample from.